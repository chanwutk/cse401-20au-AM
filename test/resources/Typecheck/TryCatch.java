class TryCatchAlreadyDefined {
  public static void main(String[] args) {
    System.out.println(new C1().run());
  }
}

class C0 extends RuntimeException {}
class C2 extends C0 {}
class C1 {
  int a;
  public int run() {
    int b;
    try {

    } catch (C0 a) {
      a = 0;
    }

    try {}
    catch (C2 b) {
      b = 0;
    }

    try{}
    catch (C0 c) {
      throw c;
    }

    try{}
    catch (C2 e) {
      throw new C0();
    }

    try{}
    catch (C1 e) {
      throw new C1();
    }

    try{}
    catch (C2 e) {
      throw new C2();
    }

    return b;
  }
}
