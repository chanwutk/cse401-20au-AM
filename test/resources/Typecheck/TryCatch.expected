TryCatch.java:35: error: incompatible types: C1 cannot be converted to Throwable
TryCatch.java:16: error: incompatible types: int cannot be converted to C0
TryCatch.java:21: error: incompatible types: int cannot be converted to C2
TryCatch.java:36: error: incompatible types: C1 cannot be converted to Throwable
